### PR TITLE
Refactor manifest

### PR DESCRIPTION
**Pull request by gboutry**

### CODE REVIEW

File: sunbeam-python/sunbeam/core/manifest.py [hunk 1]
```diff
@@ -167,48 +134,147 @@ def merge(self, other: "SoftwareConfig") -> "SoftwareConfig":
...
+        nic: str | None = None
+        cidr: str | None = None
+        gateway: str | None = None
+        start: str | None = None
+        end: str | None = None
+        network_type: typing.Literal["vlan", "flat"] | None = None
+        segmentation_id: int | None = None
+
+    class _HostMicroCephConfig(pydantic.BaseModel):
+        osd_devices: str | None = None
```
Comment (hemanthnakkina) [inline sunbeam-python/sunbeam/core/manifest.py:201]: `osd_devices` can be list as well.

**Review by hemanthnakkina — CHANGES_REQUESTED** [review]
There are certain cases like [1] which are not attributes of Manifest classes. Would you like to handle them differently or would like to leave as it is?

[1] https://github.com/canonical/snap-openstack/blob/main/sunbeam-python/sunbeam/versions.py#L153

File: sunbeam-python/sunbeam/provider/local/deployment.py [hunk 2]
```diff
@@ -71,11 +70,14 @@ class LocalDeployment(Deployment):
     def __init__(self, **data):
         super().__init__(**data)
         if self.juju_account is None:
-            self.juju_account = self._load_juju_account()
+            # self.juju_account = self._load_juju_account()
+            self.juju_account = None
```
Comment (hemanthnakkina) [inline sunbeam-python/sunbeam/provider/local/deployment.py:74]: If juju account is not getting loaded here, we can remove the lines from L#72 - L#80

Comment (gboutry) [reply sunbeam-python/sunbeam/provider/local/deployment.py:74]: This was an oversight, disabled for local testing, I've put it back

File: sunbeam-python/sunbeam/core/manifest.py (hunk 1 above)
Comment (gboutry) [reply sunbeam-python/sunbeam/core/manifest.py:201]: This is changing the behavior of manifest, and I would like to do these kind of changes in a more targeted commit.

**Review by hemanthnakkina — APPROVED** [review]

**Pull request merged**